### PR TITLE
PSEC-1416 add compliance-dataviz pipeline

### PR DESCRIPTION
**Pull request by anthonymade**
add ecs_task_pipeline pipeline type

use different roles with limited permissions for steps that interact
with different aws services

### CODE REVIEW

File: modules/ecs_task_pipeline/github_webhook.tf [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  configuration {
+    url          = aws_codepipeline_webhook.default.url
+    content_type = "json"
+    insecure_ssl = false
+    secret       = random_password.webhook.result
+  }
+
+  events = ["push"]
+}
```
Comment (cob16) [inline modules/ecs_task_pipeline/github_webhook.tf:40]: How does this work auth wise? Do I need to be Github admin to run terraform apply?

File: modules/update_task_definition_step/assets/buildspec-deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+          --query '{  containerDefinitions: taskDefinition.containerDefinitions,
+                      family: taskDefinition.family,
+                      taskRoleArn: taskDefinition.taskRoleArn,
+                      executionRoleArn: taskDefinition.executionRoleArn,
+                      networkMode: taskDefinition.networkMode,
+                      volumes: taskDefinition.volumes,
+                      placementConstraints: taskDefinition.placementConstraints,
+                      requiresCompatibilities: taskDefinition.requiresCompatibilities,
+                      cpu: taskDefinition.cpu,
+                      memory: taskDefinition.memory}' | jq '.containerDefinitions[0].image|="'${NEW_IMAGE}\")
```
Comment (cob16) [inline modules/update_task_definition_step/assets/buildspec-deploy.yaml:33]: Wow, this replaces a whole tool I used to use for this :+1:

File: main.tf [hunk 3]
```diff
@@ -142,3 +143,22 @@ module "github_webhook_report" {
...
+  source = "modules/ecs_task_pipeline"
+
+  pipeline_name = "compliance-dataviz"
+  src_repo      = "platsec-compliance-dataviz"
+  branch        = "main"
+
+  ecr_name     = "platsec-compliance-dataviz"
+  cluster_name = "security_reports_frontend"
+  service_name = "compliance_dataviz"
+  task_name    = "compliance_dataviz_task"
```
[the remark this answers is not in the document]
Comment (anthonymade) [reply main.tf:156]: Disagree. A service can have many tasks. How would that work?

[the remark this answers is not in the document]
Comment (cob16) [reply main.tf:156]: I haven't had to do that before, but I agree if that is the logic

**Review by cob16 — APPROVED** [review]

**Pull request merged**